packages/vitest/src/node/core.ts

```diff
@@ -738,6 +741,13 @@ export class Vitest {
     return files
   }
 
+  private async getProjectsByFiles(files: string[]) {
+    return await Promise.all(this.projects.filter(async (project) => {
```

Comment:
This seems very expensive. Why do we need to scan all directories when we already know the files? We can just check them against `.include` globs.

Reply:
Looks like there is already `this.projects` in the core itself. I'll change this to use it directly.